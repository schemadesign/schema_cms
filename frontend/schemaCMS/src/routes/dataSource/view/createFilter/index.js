export { default as CreateFilter } from './createFilter.container';
