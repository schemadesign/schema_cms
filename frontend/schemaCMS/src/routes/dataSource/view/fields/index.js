export { Fields } from './fields.component';
