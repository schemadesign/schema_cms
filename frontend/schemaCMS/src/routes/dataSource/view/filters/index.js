export { Filters } from './filters.component';
