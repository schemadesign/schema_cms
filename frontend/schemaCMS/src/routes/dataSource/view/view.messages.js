/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'dataSource.view.title',
    defaultMessage: 'Create Data Source',
  },
  next: {
    id: 'dataSource.view.next',
    defaultMessage: 'Next',
  },
  back: {
    id: 'dataSource.view.back',
    defaultMessage: '{cancel, select, true {Cancel} other {Back}}',
  },
  source: {
    id: 'dataSource.view.source',
    defaultMessage: 'Source',
  },
  fields: {
    id: 'dataSource.view.fields',
    defaultMessage: 'Fields',
  },
  dataWrangling: {
    id: 'dataSource.view.dataWrangling',
    defaultMessage: 'Data Wrangling',
  },
  removeDataSource: {
    id: 'dataSource.view.removeDataSource',
    defaultMessage: 'Remove Data Source',
  },
  dataSourceList: {
    id: 'dataSource.view.dataSourceList',
    defaultMessage: 'Data source list',
  },
});
