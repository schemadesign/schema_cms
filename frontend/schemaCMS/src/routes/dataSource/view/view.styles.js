import styled from 'styled-components';

export const Container = styled.div`
  padding: 0 20px;
`;

export const StepperContainer = styled.div`
  display: flex;
  width: 100%;
  margin-top: 23px;
`;

export const stepperStyles = {
  margin: '0 auto',
};

export const stepperBlockStyles = {
  pointerEvents: 'none',
};
