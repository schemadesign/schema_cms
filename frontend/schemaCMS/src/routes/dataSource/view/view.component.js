import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import { always, cond, equals, T } from 'ramda';

import { Container } from './view.styles';
import messages from './view.messages';
import { Source } from './source';
import { DataPreview } from '../../../shared/components/dataPreview';
import { DataWranglingScripts } from './dataWranglingScripts';
import { DataWranglingResult } from '../../dataWranglingResult';
import { renderWhenTrue } from '../../../shared/utils/rendering';
import { TopHeader } from '../../../shared/components/topHeader';
import {
  DATA_WRANGLING_STEP,
  DATA_WRANGLING_RESULT_STEP,
  FIELDS_STEP,
  INITIAL_STEP,
  STATUS_DRAFT,
} from '../../../modules/dataSource/dataSource.constants';

export class View extends PureComponent {
  static propTypes = {
    dataSource: PropTypes.object.isRequired,
    dataWranglingScripts: PropTypes.array.isRequired,
    fetchDataSource: PropTypes.func.isRequired,
    unmountDataSource: PropTypes.func.isRequired,
    removeDataSource: PropTypes.func.isRequired,
    fetchDataWranglingScripts: PropTypes.func.isRequired,
    uploadScript: PropTypes.func.isRequired,
    sendUpdatedDataWranglingScript: PropTypes.func.isRequired,
    intl: PropTypes.object.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }),
    match: PropTypes.shape({
      params: PropTypes.shape({
        dataSourceId: PropTypes.string.isRequired,
        step: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  };

  componentDidMount() {
    if (!this.props.dataSource.id) {
      const { dataSourceId } = this.props.match.params;

      this.props.fetchDataSource({ dataSourceId });
    }
  }

  componentWillUnmount() {
    this.props.unmountDataSource();
  }

  getHeaderSubtitle = cond([
    [equals(INITIAL_STEP), always(this.props.intl.formatMessage(messages.source))],
    [equals(FIELDS_STEP), always(this.props.intl.formatMessage(messages.fields))],
    [equals(DATA_WRANGLING_STEP), always(this.props.intl.formatMessage(messages.dataWrangling))],
    [equals(DATA_WRANGLING_RESULT_STEP), always(this.props.intl.formatMessage(messages.dataWranglingResult))],
    [T, always(null)],
  ]);

  getTitle = intl =>
    this.props.dataSource.status === STATUS_DRAFT ? intl.formatMessage(messages.title) : this.props.dataSource.name;

  getHeaderAndMenuConfig = activeStep => {
    const { dataSource, removeDataSource, intl } = this.props;
    const headerTitle = this.getTitle(intl);
    const { project: projectId, id: dataSourceId } = dataSource;

    const secondaryMenuItems = [
      {
        label: intl.formatMessage(messages.dataSourceList),
        to: `/project/${projectId}/datasource/`,
      },
      {
        label: intl.formatMessage(messages.removeDataSource),
        onClick: () => removeDataSource({ projectId, dataSourceId }),
      },
    ];

    return {
      headerTitle,
      headerSubtitle: this.getHeaderSubtitle(activeStep),
      secondaryMenuItems,
    };
  };

  renderContentForm = ({ activeStep, ...props }) =>
    cond([
      [equals(INITIAL_STEP), always(<Source {...props} />)],
      [equals(FIELDS_STEP), always(<DataPreview {...props} />)],
      [equals(DATA_WRANGLING_STEP), always(<DataWranglingScripts {...props} />)],
      [equals(DATA_WRANGLING_RESULT_STEP), always(<DataWranglingResult {...props} />)],
      [equals(5), always(null)],
      [equals(6), always(null)],
      [T, always(null)],
    ])(activeStep);

  renderContent = renderWhenTrue(() => {
    const {
      dataSource,
      intl,
      match: {
        params: { step },
      },
    } = this.props;
    const activeStep = parseInt(step, 10);
    const topHeaderConfig = this.getHeaderAndMenuConfig(activeStep);

    return (
      <Fragment>
        <TopHeader {...topHeaderConfig} />
        {this.renderContentForm({
          activeStep,
          intl,
          dataSource,
          ...this.props,
        })}
      </Fragment>
    );
  });

  render() {
    return <Container>{this.renderContent(!!this.props.dataSource.id)}</Container>;
  }
}
