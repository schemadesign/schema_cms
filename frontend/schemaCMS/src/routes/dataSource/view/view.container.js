import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import { View } from './view.component';
import {
  DataSourceRoutines,
  selectDataSource,
  selectPreviewData,
  selectIsAnyJobProcessing,
} from '../../../modules/dataSource';
import {
  DataWranglingScriptsRoutines,
  selectDataWranglingScripts,
  selectCustomScripts,
  selectImageScrapingFields,
} from '../../../modules/dataWranglingScripts';
import { selectFilters } from '../../../modules/filter';
import { FilterRoutines } from '../../../modules/filter/filter.redux';
import { selectIsAdmin } from '../../../modules/userProfile';

const mapStateToProps = createStructuredSelector({
  dataSource: selectDataSource,
  dataWranglingScripts: selectDataWranglingScripts,
  filters: selectFilters,
  previewData: selectPreviewData,
  isAnyJobProcessing: selectIsAnyJobProcessing,
  isAdmin: selectIsAdmin,
  customScripts: selectCustomScripts,
  imageScrapingFields: selectImageScrapingFields,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      removeDataSource: promisifyRoutine(DataSourceRoutines.removeOne),
      onDataSourceChange: promisifyRoutine(DataSourceRoutines.updateOne),
      fetchDataWranglingScripts: promisifyRoutine(DataWranglingScriptsRoutines.fetchList),
      uploadScript: promisifyRoutine(DataWranglingScriptsRoutines.uploadScript),
      sendUpdatedDataWranglingScript: promisifyRoutine(DataWranglingScriptsRoutines.sendList),
      fetchPreview: promisifyRoutine(DataSourceRoutines.fetchPreview),
      fetchFilters: promisifyRoutine(FilterRoutines.fetchList),
      setFilters: promisifyRoutine(FilterRoutines.setFilters),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  injectIntl,
  withRouter
)(View);
