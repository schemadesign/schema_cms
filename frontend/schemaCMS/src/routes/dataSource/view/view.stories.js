import React from 'react';
import { storiesOf } from '@storybook/react';

import { View } from './view.component';

export const defaultProps = {
  dataSource: {
    project: '1',
    metaData: {},
  },
  previewData: {},
  dataWranglingScripts: [],
  filters: [],
  isAnyJobProcessing: false,
  fetchFields: Function.prototype,
  setFilters: Function.prototype,
  fetchFilters: Function.prototype,
  unmountFields: Function.prototype,
  fetchDataSource: Function.prototype,
  removeDataSource: Function.prototype,
  fetchPreview: Function.prototype,
  fetchDataWranglingScripts: Function.prototype,
  uploadScript: Function.prototype,
  sendUpdatedDataWranglingScript: Function.prototype,
  intl: {
    formatMessage: ({ defaultMessage }) => defaultMessage,
  },
  history: {
    push: Function.prototype,
  },
  match: {
    params: {
      dataSourceId: '1',
      step: '1',
    },
  },
};

storiesOf('DataSource/View', module).add('Default', () => <View {...defaultProps} />);
