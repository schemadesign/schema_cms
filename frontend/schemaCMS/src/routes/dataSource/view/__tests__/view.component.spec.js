import React from 'react';
import { shallow } from 'enzyme';

import { View } from '../view.component';
import { defaultProps } from '../view.stories';
import { STATUS_DRAFT } from '../../../../modules/dataSource/dataSource.constants';

describe('View: Component', () => {
  const component = props => <View {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should not render content', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render source step', () => {
    const props = {
      dataSource: {
        id: 'id',
        name: 'name',
        project: '1',
      },
    };
    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render second step', () => {
    const props = {
      dataSource: {
        id: 'id',
        name: 'name',
        step: '2',
        project: '1',
      },
    };
    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should not render any step', () => {
    const props = {
      dataSource: {
        id: 'id',
        name: 'name',
        project: '1',
      },
      match: {
        params: {
          projectId: '1',
          dataSourceId: '1',
          step: '0',
        },
        url: '/project/view/1/datasource/view/1/0',
      },
    };

    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render draft', () => {
    const props = {
      dataSource: {
        id: 'id',
        status: STATUS_DRAFT,
        project: '1',
      },
    };

    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });
});
