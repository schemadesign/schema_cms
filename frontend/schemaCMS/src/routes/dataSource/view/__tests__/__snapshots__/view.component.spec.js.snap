// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View: Component should not render any step 1`] = `
<styled.div>
  <TopHeader
    headerSubtitle={null}
    headerTitle="name"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Data source list",
          "to": "/project/view/1/datasource/list",
        },
        Object {
          "label": "Remove Data Source",
          "onClick": [Function],
        },
      ]
    }
  />
  <NavigationContainer
    right={false}
  >
    <BackButton
      onClick={[Function]}
      title="Back"
    />
    <NextButton
      onClick={[Function]}
    />
    <styled.div>
      <Component
        activeStep={0}
        customStyles={
          Object {
            "margin": "0 auto",
          }
        }
        onStepChange={[Function]}
        steps={6}
      />
    </styled.div>
  </NavigationContainer>
</styled.div>
`;

exports[`View: Component should not render content 1`] = `<styled.div />`;

exports[`View: Component should render draft 1`] = `
<styled.div>
  <TopHeader
    headerSubtitle="Source"
    headerTitle="Create Data Source"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Data source list",
          "to": "/project/view/1/datasource/list",
        },
        Object {
          "label": "Remove Data Source",
          "onClick": [Function],
        },
      ]
    }
  />
  <WithTheme(SourceComponent)
    bindSubmitForm={[Function]}
    dataSource={
      Object {
        "id": "id",
        "status": "draft",
      }
    }
    dataWranglingScripts={Array []}
    fetchDataSource={[Function]}
    fetchDataWranglingScripts={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
          "projectId": "1",
          "step": "1",
        },
        "url": "/project/view/1/datasource/view/1/1",
      }
    }
    removeDataSource={[Function]}
    sendUpdatedDataWranglingScript={[Function]}
    unmountDataSource={[Function]}
    updateDataSource={[Function]}
    uploadScript={[Function]}
  />
  <NavigationContainer
    right={false}
  >
    <BackButton
      onClick={[Function]}
      title="Cancel"
    />
    <NextButton
      onClick={[Function]}
    />
    <styled.div>
      <Component
        activeStep={1}
        customStyles={
          Object {
            "margin": "0 auto",
            "pointerEvents": "none",
          }
        }
        onStepChange={[Function]}
        steps={6}
      />
    </styled.div>
  </NavigationContainer>
</styled.div>
`;

exports[`View: Component should render second step 1`] = `
<styled.div>
  <TopHeader
    headerSubtitle="Source"
    headerTitle="name"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Data source list",
          "to": "/project/view/1/datasource/list",
        },
        Object {
          "label": "Remove Data Source",
          "onClick": [Function],
        },
      ]
    }
  />
  <WithTheme(SourceComponent)
    bindSubmitForm={[Function]}
    dataSource={
      Object {
        "id": "id",
        "name": "name",
        "step": "2",
      }
    }
    dataWranglingScripts={Array []}
    fetchDataSource={[Function]}
    fetchDataWranglingScripts={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
          "projectId": "1",
          "step": "1",
        },
        "url": "/project/view/1/datasource/view/1/1",
      }
    }
    removeDataSource={[Function]}
    sendUpdatedDataWranglingScript={[Function]}
    unmountDataSource={[Function]}
    updateDataSource={[Function]}
    uploadScript={[Function]}
  />
  <NavigationContainer
    right={false}
  >
    <BackButton
      onClick={[Function]}
      title="Cancel"
    />
    <NextButton
      onClick={[Function]}
    />
    <styled.div>
      <Component
        activeStep={1}
        customStyles={
          Object {
            "margin": "0 auto",
          }
        }
        onStepChange={[Function]}
        steps={6}
      />
    </styled.div>
  </NavigationContainer>
</styled.div>
`;

exports[`View: Component should render source step 1`] = `
<styled.div>
  <TopHeader
    headerSubtitle="Source"
    headerTitle="name"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Data source list",
          "to": "/project/view/1/datasource/list",
        },
        Object {
          "label": "Remove Data Source",
          "onClick": [Function],
        },
      ]
    }
  />
  <WithTheme(SourceComponent)
    bindSubmitForm={[Function]}
    dataSource={
      Object {
        "id": "id",
        "name": "name",
      }
    }
    dataWranglingScripts={Array []}
    fetchDataSource={[Function]}
    fetchDataWranglingScripts={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
          "projectId": "1",
          "step": "1",
        },
        "url": "/project/view/1/datasource/view/1/1",
      }
    }
    removeDataSource={[Function]}
    sendUpdatedDataWranglingScript={[Function]}
    unmountDataSource={[Function]}
    updateDataSource={[Function]}
    uploadScript={[Function]}
  />
  <NavigationContainer
    right={false}
  >
    <BackButton
      onClick={[Function]}
      title="Cancel"
    />
    <NextButton
      onClick={[Function]}
    />
    <styled.div>
      <Component
        activeStep={1}
        customStyles={
          Object {
            "margin": "0 auto",
          }
        }
        onStepChange={[Function]}
        steps={6}
      />
    </styled.div>
  </NavigationContainer>
</styled.div>
`;
